gc.md

```diff
@@ -189,53 +226,133 @@ ToSpace starts as unallocated memory.
 
 - now we can allocate E
 
-#### Write Barriers
+#### Considerations
 
-[read](http://jayconrod.com/posts/55/a-tour-of-v8-garbage-collection) *Write barriers: the secret ingredient*
+[watch](https://youtu.be/VhpdsjBUS3g?t=15m30s)
 
-[barrier](http://www.memorymanagement.org/glossary/b.html#term-barrier-1) | [write
-barrier](http://www.memorymanagement.org/glossary/w.html#term-write-barrier) | [read
-barrier](http://www.memorymanagement.org/glossary/r.html#term-read-barrier)
+- every allocation brings us closer to GC pause
+- even though as many steps of collection are performed in parallel, **every collection pauses our
+  app**
+- try to pre-alloc as much as possible ahead of time
```

Comment:
Pre-allocation as much as possible is in general not a good idea :)
GC complexity depends on the live set. Pre-allocation artificially increases the live set.

Reply:
I was talking about preallocating things that you'll use a lot, but they don't change, i.e.:

```js
// msg allocated one time, never needs garbage collecting
const msg = 'User not found'

function onuserNotFound() {
  emit(msg)
}
```

Isn't that more efficient than:

```js
function onuserNotFound() {
  // message allocated for each invocation and will be collected over and over
  emit('User not found')
}
```

Reply:
Also I'm bit unclear WRT to (pre)allocating + immediately filling a large Array and performance.
It used to be that you'd want to pre-alloc as long as it had `<100K` items, in the below example function _one_ performed better than _two_.

Is this still true?

```js
function one(items) {
  const result = new Array(items.length)
  for (var i = 0; i < items.length; i++) {
    result[i] = items[i] * 10
  }
  return result
}

function two(items) {
  const result = []
  for (const item of items) {
    result.push(item * 10)
  }
  return result
}
```

Reply:
Tried to address in 9d79c90, but still would like clarifications about my questions here, so I can add that info.

Reply:
It depends. Preallocating immutable objects used in a hot loop is probably worthwhile.

Preallocating mutable object that refers to other objects or closures increases chances of memory leaks because now the developer has to manually delete unused references. Memory leaks increase memory usage and GC pauses.

Also a preallocated object slows down all subsequent GCs because each GC will have to visit it.

Reply:
Yes, that's correct. Not sure about the exact limits. I think @mathiasbynens recently checked that